### PR TITLE
#498 BULK - DEU-MATCH CLEARED - bug reported

### PR DESCRIPTION
**Pull request by MiKaylaH22**
the spreadsheet sometimes formats the income amount to remove the . and the 00 but it is not consistent.
Used the left 4 of the amount to confirm we have the right match the report (excel) comes from REPT/IEVC which makes me confident this is sufficient
example
2150.50
excel income 
2150.5

### CODE REVIEW

**Comment by MiKaylaH22** [comment]
Ran again and cleared the list without issue

File: deu/bulk-match-cleared.vbs [hunk 1]
```diff
@@ -325,10 +323,13 @@ Do 'purpose is to read each excel row and to add into each excel array '
 	   	match_based_array(maxis_case_number_const,  entry_record)	 = MAXIS_case_number
 	   	match_based_array(client_ssn_const, 		entry_record)	 = trim(replace(objExcel.cells(excel_row, excel_col_client_ssn), "-", ""))
 		match_based_array(program_const,  			entry_record)    = trim(objExcel.cells(excel_row, excel_col_program).Value)
-	   	match_based_array(amount_const, 			entry_record) 	 = trim(objExcel.cells(excel_row, excel_col_amount).Value)
-		match_based_array(amount_const, 		    entry_record)    = replace(objExcel.cells(excel_row, excel_col_amount).Value, "$", "")
+        'FormatNumber(number, deciaml places, leading zero(false), negative number (false), use deliminator(false))
+        match_based_array(amount_const, 		    entry_record)    = FormatNumber(match_based_array(amount_const, entry_record), 2, 0, 0, 0) 'this is formating to help the script read the number as a number'
+        match_based_array(amount_const, 			entry_record) 	 = trim(objExcel.cells(excel_row, excel_col_amount).Value)
+        match_based_array(amount_const, 		    entry_record)    = replace(objExcel.cells(excel_row, excel_col_amount).Value, "$", "")
```
Comment (C-Love) [inline deu/bulk-match-cleared.vbs:329]: This replacing situation makes line 328 unnecessary. You can replace the already defined variable, or you can defined the variable and do the 2 lines of replacing at the same time.

File: deu/bulk-match-cleared.vbs [hunk 2]
```diff
@@ -325,10 +323,13 @@ Do 'purpose is to read each excel row and to add into each excel array '
...
 		match_based_array(amount_const, 		    entry_record)    = replace(objExcel.cells(excel_row, excel_col_amount).Value, ",", "")
-	   	match_based_array(income_source_const, 		entry_record)    = trim(objExcel.cells(excel_row, excel_col_income_source).Value)
+        match_based_array(amount_const, 		    entry_record)    = match_based_array(amount_const, 		    entry_record) *1 'this is so the amount wil be read as a number'
```
Comment (C-Love) [inline deu/bulk-match-cleared.vbs:331]: Recommend:


        match_based_array(amount_const, 			entry_record) 	 = trim(objExcel.cells(excel_row, excel_col_amount).Value)
        match_based_array(amount_const, 		    entry_record)    = replace(match_based_array(amount_const, entry_record), "$", "")
        match_based_array(amount_const, 		    entry_record)    = replace(match_based_array(amount_const, entry_record), ",", "")
        match_based_array(amount_const, 		    entry_record)    = FormatNumber(match_based_array(amount_const, entry_record), 2, 0, 0, 0) 'this is formating to help the script read the number as a number'
        match_based_array(amount_const, 		    entry_record)    = match_based_array(amount_const, 		    entry_record) *1 'this is so the amount wil be read as a number'

**Review by C-Love — APPROVED** [review]
Looks good.

**Pull request merged**